### PR TITLE
Improve solver code and docs

### PR DESCRIPTION
**Pull request by Seanny123**
**Motivation and context:**
Fix some inconsistencies and confusion that can occur when reading the `nengo.utils.least_squares_solvers` code, as described in #1229 and #1065 

**How has this been tested?**
No functional changes, so as long as all the normal tests are passed, this code should be fine.

**How long should this take to review?**

- Quick (less than 40 lines changed or changes are straightforward)

**Types of changes:**

- Bug fix (non-breaking change which fixes an issue)
- Non-code change (touches things like tests, documentation, build scripts)

**Checklist:**

- [x] I have read the **CONTRIBUTING.rst** document.
- [x] I have updated the documentation accordingly.
- [x] All new and existing tests passed.

### CODE REVIEW

File: nengo/utils/least_squares_solvers.py [hunk 1]
```diff
@@ -67,7 +67,7 @@ def __call__(self, A, y, sigma, rng=None):
             x = np.dot(L, np.dot(L.T, b))
 
         x = np.dot(A.T, x) if transpose else x
```
Comment (hunse) [inline nengo/utils/least_squares_solvers.py]: ``x`` should also change to ``X``, to be consistent with the others.

**Comment by Seanny123** [comment]
@hunse fixed it?

**Comment by hunse** [comment]
Yeah, looks great.

**Comment by tbekolay** [comment]
@hunse could you file an approved review if this looks ready to merge for you? (Go to Files Changed, click "Review changes" at the top right > check Approved > Submit, no text required)

**Comment by hunse** [comment]
I royally screwed up GitHub, though. Now what's shown here doesn't match what's in git.

**Comment by tbekolay** [comment]
> Now what's shown here doesn't match what's in git.

Curious! I like a challenge ;) Will merge this in shortly.

**Pull request merged**